compiler/rustc_pattern_analysis/src/lints.rs

```diff
@@ -27,11 +28,11 @@ use crate::MatchArm;
 ///
 /// This is not used in the main algorithm; only in lints.
 #[derive(Debug)]
-pub(crate) struct PatternColumn<'p, 'tcx> {
-    patterns: Vec<&'p DeconstructedPat<'p, 'tcx>>,
+pub(crate) struct PatternColumn<'a, 'p, 'tcx> {
```

Comment:
Is the `'p` lifetime now invariant? Why did you need to introduce a new lifetime here?

Reply:
Yeah, it's invariant because `SmallVec<[T; N]>` is invariant in `T`

Reply:
Honestly I'm not exactly sure, I was happy to find something that worked, this took a while to untangle

Reply:
Darn